service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java

```diff
@@ -245,6 +254,20 @@ public HearingBooking getHearingBooking(UUID id, List<Element<HearingBooking>> h
         caseFields.put("preHearingAttendanceDetails", hearingBooking.getPreAttendanceDetails());
         caseFields.put("sendNoticeOfHearingTranslationRequirements", hearingBooking.getTranslationRequirements());
 
+        if (YES.getValue().equals(hearingBooking.getEndDateDerived())) {
+            if (hearingBooking.getHearingDays() != null) {
+                caseFields.put(HEARING_DURATION, DAYS.getType());
+                caseFields.put(HEARING_DAYS, hearingBooking.getHearingDays());
+            } else if (hearingBooking.getHearingHours() != null || hearingBooking.getHearingMinutes() != null) {
+                caseFields.put(HEARING_DURATION, HOURS_MINS.getType());
+                caseFields.put(HEARING_HOURS, hearingBooking.getHearingHours());
+                caseFields.put(HEARING_MINUTES, hearingBooking.getHearingMinutes());
+            }
+        } else if (hearingBooking.getEndDate() != null) {
+            caseFields.put(HEARING_DURATION, HearingDuration.DATE_TIME.getType());
```

Comment:
static import

Reply:
Can't do static import as it clashes with DateFormatterHelper.DATE_TIME.